### PR TITLE
ocamltest: fix bug in stub code

### PR DESCRIPTION
**Pull request by shindere**
ocamltest did not compile when -safe-string was enabled.
gcc rightfully gave a warning in this case.

This PR fixes this.

Should be included in 4.06.

### CODE REVIEW

File: ocamltest/run_stubs.c [hunk 1]
```diff
@@ -80,5 +82,7 @@ CAMLprim value caml_run_command(value caml_settings)
   settings.logger = logToChannel;
   settings.loggerData = Channel(Field(caml_settings, 7));
   res = run_command(&settings);
+  for (p = *(settings.argv); p!=NULL; p++)
+    caml_stat_free(p);
```
Comment (nojb) [inline ocamltest/run_stubs.c]: You should also free `settings.argv` itself, right ? (compare with https://github.com/ocaml/ocaml/pull/1357/files#diff-9861eb39c9e039fa7cb0230fc8e2a6ca).

**Comment by shindere** [comment]
Indeed, thanks a lot @nojb.

Also, the original code was actually wrong, as mentionned offline by
@damiendoligez.

Hopefully this one is okay.

**Review by damiendoligez — APPROVED** [review]
Now it's OK.

**Comment by damiendoligez** [comment]
Merged to trunk. Don't forget to cherry-pick to 4.06.

**Comment by shindere** [comment]
Damien Doligez (2017/09/22 13:27 +0000):
> Merged to trunk.

Thanks, and thanks for the reviews.

> Don't forget to cherry-pick to 4.06.

Done. See a3757857901ca0157657ee4701c27b3fd586091b

**Pull request merged**